Mastodon/Scene/Thread/MastodonStatusThreadViewModel+State.swift

```diff
@@ -0,0 +1,162 @@
+// Copyright © 2024 Mastodon gGmbH. All rights reserved.
+
+import Foundation
+import MastodonSDK
+
+extension MastodonStatusThreadViewModel {
+    // Bookmark
+    func handleBookmark(_ status: MastodonStatus) {
+        ancestors = handleBookmark(status, items: ancestors)
+        descendants = handleBookmark(status, items: descendants)
+    }
+    
+    private func handleBookmark(_ status: MastodonStatus, items: [StatusItem]) -> [StatusItem] {
+        var newRecords = Array(items)
+        guard let index = newRecords.firstIndex(where: { $0.mastodonStatus?.id == status.id }) else {
+            return items
+        }
+        var newRecord = newRecords[index]
+        newRecord.mastodonStatus = status
+        newRecords[index] = newRecord
+        return newRecords
+    }
+    
+    // Favorite
+    func handleFavorite(_ status: MastodonStatus) {
+        ancestors = handleFavorite(status, items: ancestors)
+        descendants = handleFavorite(status, items: descendants)
+    }
+    
+    private func handleFavorite(_ status: MastodonStatus, items: [StatusItem]) -> [StatusItem] {
+        var newRecords = Array(items)
+        guard let index = newRecords.firstIndex(where: { $0.mastodonStatus?.id == status.id }) else {
+            return items
+        }
+        var newRecord = newRecords[index]
+        newRecord.mastodonStatus = status
+        newRecords[index] = newRecord
+        return newRecords
+    }
+    
+    // Reblog
+    func handleReblog(_ status: MastodonStatus, _ isReblogged: Bool) {
+        ancestors = handleReblog(status, isReblogged, items: ancestors)
+        descendants = handleReblog(status, isReblogged, items: descendants)
+    }
+    
+    private func handleReblog(_ status: MastodonStatus, _ isReblogged: Bool, items: [StatusItem]) -> [StatusItem] {
+        var newRecords = Array(items)
+        switch isReblogged {
+        case true:
+            let index: Int
+            if let idx = newRecords.firstIndex(where: { $0.mastodonStatus?.id == status.id }) {
+                index = idx
+            } else if let idx = newRecords.firstIndex(where: { $0.mastodonStatus?.reblog?.id == status.reblog?.id }) {
+                index = idx
+            } else {
+                return newRecords
+            }
+            var newRecord = newRecords[index]
+            newRecord.mastodonStatus = status
+            newRecords[index] = newRecord
+            
+        case false:
+            let index: Int
+            if let idx = newRecords.firstIndex(where: { $0.mastodonStatus?.id == status.id }) {
+                index = idx
+            } else if let idx = newRecords.firstIndex(where: { $0.mastodonStatus?.reblog?.id == status.reblog?.id }) {
+                index = idx
+            } else {
+                return newRecords
+            }
+            var newRecord = newRecords[index]
+            newRecord.mastodonStatus = status
+            newRecords[index] = newRecord
+        }
```

Comment:
Where's the difference between these two `case`?

Reply:
Hm, there was a difference at some point, but from a logical perspective it doesn't make much sense, looks like this can go away now 👍